pandas/_libs/tslibs/conversion.pyx

```diff
@@ -222,7 +222,7 @@ def ensure_datetime64ns(arr: ndarray, copy: bool=True):
         dtype = arr.dtype
         arr = arr.astype(dtype.newbyteorder("<"))
 
-    ivalues = arr.view(np.int64).ravel("K")
+    ivalues = arr.view(np.int64).ravel("C")
```

Comment:
this is going to make a copy if arr has order="F" isnt it?

Reply:
Yeah unfortunately, but not sure how we can avoid this here. Could transpose back before passing in, but I think that is not an ideal solution

Reply:
could arr.flat be useful here?